apps/designer/app/designer/features/sidebar-left/panels/pages/index.tsx

```diff
@@ -0,0 +1,100 @@
+// @todo: consider moving code from index.tsx to separate files similar to other panels
+
+import { TreeNodeLabel, TreeNode } from "@webstudio-is/design-system";
+import { type Publish } from "~/shared/pubsub";
+import { StackIcon } from "@webstudio-is/icons";
+import type { TabName } from "../../types";
+import { Header } from "../../lib/header";
+import { type Page, type Pages, type Project } from "@webstudio-is/project";
+import { useMemo } from "react";
+import { useNavigate } from "react-router-dom";
+import { type Config } from "~/config";
+
+type TabContentProps = {
+  onSetActiveTab: (tabName: TabName) => void;
+  publish: Publish;
+  project: Project;
+  pages: Pages;
+  currentPageId: string;
+  config: Config;
+};
+
+type PagesTreeNode =
+  | {
+      // currently used only for root node
+      type: "folder";
+      id: string;
+      children: PagesTreeNode[];
+    }
+  | {
+      type: "page";
+      id: string;
+      data: Page;
+    };
+
+const toTreeData = (pages: Pages): PagesTreeNode => {
+  return {
+    type: "folder",
+    id: "root",
+    children: [pages.homePage, ...pages.pages].map((data) => ({
+      type: "page",
+      id: data.id,
+      data,
+    })),
+  };
+};
+
+const staticTreeProps = {
+  getItemChildren(node: PagesTreeNode) {
+    if (node.type === "folder") {
+      return node.children;
+    }
+    return [];
+  },
+  getIsExpanded(_node: PagesTreeNode) {
+    return true;
+  },
+  renderItem(props: { data: PagesTreeNode; isSelected: boolean }) {
+    if (props.data.type === "folder") {
+      return null;
+    }
+
+    return (
+      <TreeNodeLabel
+        isSelected={props.isSelected}
+        text={props.data.data.name}
+      />
+    );
+  },
+};
+
+export const TabContent = ({
+  onSetActiveTab,
+  project,
+  pages,
+  currentPageId,
+  config,
+}: TabContentProps) => {
+  const pagesTree = useMemo(() => pages && toTreeData(pages), [pages]);
+
+  const navigate = useNavigate();
+  const handleSelect = (pageId: string) => {
+    navigate(`${config.designerPath}/${project.id}?pageId=${pageId}`);
```

Comment:
There is this future use case that would be great to make possible: remotely connecting the designer to any site/app that uses SDK. Think of it as of chrome devtools connecting to v8 over protocol.

The question with the url's is that they are kindof owned by the application on canvas. Lets brainstorm how can we allow the application on canvas to fully own itself, including the urls.